game/hackmon.cc

```diff
@@ -0,0 +1,16 @@
+#include <vector>
+#include <string>
+#include <map>
+#include <memory>
+
+#include "hackmon.h"
+#include "debuffName.h"
+#include "statInfo.h"
+using namespace std;
+
+Hackmon::Hackmon(string hackmonName, vector<Move> moves, Stats stats, vector<Family> family):
+name{hackmonName}, moves{moves}, stats{stats}, family{family} {
+  currentMove = nullptr;
```

Comment:
`currentMove`, `inBattle`, and `debuff` should be set in the MIL. Also you're missing `mobility`.